src/geometry/manifold/derivation_bundle.lean

```diff
@@ -0,0 +1,165 @@
+/-
+Copyright © 2020 Nicolò Cavalleri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nicolò Cavalleri
+-/
+
+import geometry.manifold.algebra.smooth_functions
+import ring_theory.derivation
+
+/-!
+
+# Derivation bundle
+
+In this file we define the derivations at a point of a manifold on the algebra of smooth fuctions.
+Moreover we define the total bundle of derivations (although at the moment it has not been given a
+topology). Finally we define the differential of a function in terms of derivations.
+
+The content of this file is not meant to be regarded as an alternative definition to the current
+tangent bundle but rather as a purely algebraic theory that provides a purely algebraic definition
+of the Lie algebra for a Lie group.
+
+-/
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+{E : Type*} [normed_group E] [normed_space 𝕜 E]
+{H : Type*} [topological_space H] (I : model_with_corners 𝕜 E H)
+{M : Type*} [topological_space M] [charted_space H M]
+
+open_locale manifold
+
+namespace point_derivation
+
+instance smooth_functions_algebra : algebra 𝕜 C^∞⟮I, M; 𝕜⟯ := by apply_instance
+instance smooth_functions_tower : is_scalar_tower 𝕜 C^∞⟮I, M; 𝕜⟯ C^∞⟮I, M; 𝕜⟯ := by apply_instance
+
+/-- Evaluation at a point is a ring homomorphism. -/
+def smooth_funtion.eval' (x : M) : C^∞⟮I, M; 𝕜⟯ →+* 𝕜 :=
+{ to_fun    := λ f, f x,
+  map_one'  := rfl,
+  map_mul'  := λ f g, rfl,
+  map_zero' := rfl,
+  map_add'  := λ f g, rfl }
+
+/-- The above evaluation gives rise to an algebra structure of `C^∞⟮I, M; 𝕜⟯` on `𝕜`. -/
+def algebra (x : M) : algebra C^∞⟮I, M; 𝕜⟯ 𝕜 := (smooth_funtion.eval' I x).to_algebra
```

Comment:
Could you maybe expand a doc string, or even write a `/-! -/` doc about here, explaining the mathematical role of this algebra structure? It is a bit of a surprise when you first see it here.

Reply:
I believe this is just a slightly more formal way to introduce the derivation bundle that many differential geometry textbooks introduce as the tangent bundle. I am not sure that there is a lot more to say that "this is actually what is really going on when people talk about derivations in the context of the tangent bundle in differential geometry"... What should I say?